frontend/src/app/features/folios/AddToFolio.tsx

```diff
@@ -77,51 +117,52 @@ const AddToFolio: FC<AddToFolioProps> = ({
     });
   };
 
-  return (
-    <OverlayTrigger
-      trigger="click"
-      placement={popupPlacement}
-      rootClose // closes the popover on outside click
-      transition={false}
-      onToggle={(open) => {
-        if (open && checkUserAuthentication()) {
-          getFolioItems();
+  if (!showSearchBoxOnly) {
+    return (
+      <OverlayTrigger
+        trigger="click"
+        placement={popupPlacement}
+        rootClose // closes the popover on outside click
+        transition={false}
+        onToggle={(open) => {
+          if (open && checkUserAuthentication()) {
+            getFolioItems();
+          }
+        }}
+        overlay={
+          <Popover className="folio-popover">
+            {renderSearchInput()}
+          </Popover>
         }
-      }}
-      overlay={
-        <Popover className="folio-popover">
-          <SearchInput
-            loading={loading}
-            label={'Search Folios'}
-            placeHolderText={'Search Folios'}
-            searchTerm={searchTerm}
-            clearSearch={() => {
-              setSearchTerm('');
-            }}
-            handleSearchChange={(e) => {
-              setSearchTerm(e.target.value);
-            }}
-            options={searchOptions}
-            optionSelectHandler={(value) => {
-              handleFolioSelect(value);
-            }}
-          />
-        </Popover>
-      }
-    >
-      <button
-        className={clsx([
-          'search-result-actions-btn',
-          'search-result-actions-btn-highlight',
-          triggerClassName,
-        ])}
-        disabled={disabled}
       >
-        <FolderPlusIcon />
-        <span>{label}</span>
-      </button>
-    </OverlayTrigger>
+        <button
+          className={clsx([
+            'search-result-actions-btn',
+            'search-result-actions-btn-highlight',
+            triggerClassName,
+          ])}
+          disabled={disabled}
+        >
+          <FolderPlusIcon />
+          <span>{label}</span>
+        </button>
+      </OverlayTrigger>
+    );
+  } else {
+    
+  return (
+    <div className="d-flex  flex-column justify-content-center">
+        {renderSearchInput(true)}
+      <div
+        onClick={handleClose}
+        className="d-flex flex-row align-items-center pt-2 justify-content-center "      >        
+        <XmarkIcon className="custom-search-label"></XmarkIcon>
+        <span className='custom-search-label'>Close</span>
+      </div>
+    </div>
```

Comment:
So for this to look right on the page, the `AddToFolio` component has to have `ModalDialogWrapperWithHeader` as its parent, and I'm not a fan of this approach.

What I think we should do instead is to render the modal in the `overlay` props based on the screen size. The idea here is something like

```
const isSmall = useMediaQuery('@media (max-width:899.95px)'); // not actual code, please don't use this

return (
  <OverlayTrigger
      overlay={isSmall ? _ search input within modal here_ : _ regular popup _}
  />
)
```

This should also take care of my other concern: your PR changes the behavior of one "Add to Folio" button, but we have at least two more of them in the app (in Summary and Search). Placing the logic in the Add To Folio component itself saves us the trouble of modifying other components and adding more state variables in them, making it more maintainable in the long run.

Reply:
Thanks for the feedback and the pair programming we did on this component.